File: app/src/main/java/io/github/fanky10/android/mvpunittest/mvp/model/LoginModel.java
package io.github.fanky10.android.mvpunittest.mvp.model;

import android.support.annotation.NonNull;

import io.github.fanky10.android.mvpunittest.data.entities.User;
import io.github.fanky10.android.mvpunittest.data.repository.UsersRepository;
import io.github.fanky10.android.mvpunittest.mvp.LoginContract;

public class LoginModel implements LoginContract.Model {

    private final UsersRepository usersRepository;

    public LoginModel(@NonNull UsersRepository usersRepository) {
        this.usersRepository = usersRepository;
    }

    @Override
    public boolean authenticateUser(String username, String password) {
        if (usersRepository.getUsers() == null) {
            return false;
        }

        for (User user: usersRepository.getUsers()) {
            if (user.getUsername().equals(username) && user.getPassword().equals(password)) {
               return true;
            }
        }
        return false;
    }
}

File: app/src/main/java/io/github/fanky10/android/mvpunittest/mvp/model/LoginModel.kt
package io.github.fanky10.android.mvpunittest.mvp.model

import io.github.fanky10.android.mvpunittest.data.repository.UsersRepository
import io.github.fanky10.android.mvpunittest.mvp.LoginContract

class LoginModel(private val usersRepository: UsersRepository) : LoginContract.Model {

    override fun authenticateUser(username: String, password: String): Boolean {
        for (user in usersRepository.users) {
            if (user.username == username && user.password == password) {
                return true
            }
        }
        return false
    }
}
